lib/modules/manager/pub/artifacts.ts

```diff
@@ -9,18 +10,23 @@ import {
   readLocalFile,
   writeLocalFile,
 } from '../../../util/fs';
-import { regEx } from '../../../util/regex';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import type { PubspecLock, PubspecSdk } from './types';
 
-function getFlutterConstraint(lockFileContent: string): string | undefined {
-  return regEx(/^\tflutter: ['"](?<flutterVersion>.*)['"]$/m).exec(
-    lockFileContent
-  )?.groups?.flutterVersion;
-}
-
-function getDartConstraint(lockFileContent: string): string | undefined {
-  return regEx(/^\tdart: ['"](?<dartVersion>.*)['"]$/m).exec(lockFileContent)
-    ?.groups?.dartVersion;
+export function getConstraint(
+  lockFileContent: string,
+  lockFileSdkKey: keyof PubspecSdk
+): string | undefined {
+  try {
+    const data = load(lockFileContent, { json: true }) as PubspecLock;
```

Comment:
parse once with lazy, checkout npm manager for sample

Reply:
Sorry, not sure what you meant with this and couldn't find the sample, would you be able to share the link to the sample?

Reply:
ups, PR wasn't merged yet

https://github.com/renovatebot/renovate/pull/23131